src/extension.ts

```diff
@@ -69,6 +71,8 @@ export async function activate(ctx: vscode.ExtensionContext) {
   const buildDir = getBuildDirectory(sourceDir);
   workspaceState.update("mesonbuild.buildDir", buildDir);
   workspaceState.update("mesonbuild.sourceDir", sourceDir);
+  cpptools = new CpptoolsProvider(buildDir);
+  registerCppToolsProvider(ctx, cpptools);
```

Comment:
Should we only do this if it is a C or C++ project?

Reply:
I don't think it matters. If the user does not add the `"C_Cpp.default.configurationProvider": "mesonbuild.mesonbuild"` setting, then non of the configuration methods will be called. And they do, this only affects C/C++ files detected by the cpptools extension.